### PR TITLE
Assignment 3 - Bryan Snyder

### PR DESCRIPTION
**Pull request by Hex-Paladin**
website:
https://lively-sea-0f3b4921e.5.azurestaticapps.net
API:
lexiquestapi.azurewebsites.net

Still a lot of work to do on this one.

### CODE REVIEW

**Comment by alyxadarr** [comment]
I will review this!

File: Wordle.Api/Wordle.Api/Controllers/WordController.cs [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
+    public string GetRandomWord()
     {
-        [HttpGet(Name = "GetWord")]
-        public string Get()
-        {
-            return wordOfTheDayService.GetRandomWord();
-        }
+        return wordOfTheDayService.GetRandomWord();
+    }
+
```
Comment (alyxadarr) [inline Wordle.Api/Wordle.Api/Controllers/WordController.cs:14]: Be sure to add another Controller file in here for the leaderboard implementation! and another service for that too

File: .github/workflows/azure-static-web-apps-orange-forest-0eb1d671e.yml [hunk 2]
```diff
@@ -4,15 +4,18 @@ on:
   push:
     branches:
       - main
```
Comment (alyxadarr) [inline .github/workflows/azure-static-web-apps-orange-forest-0eb1d671e.yml:6]: I would add Assignment-3 so it deploys when you push on that branch

File: wordle-web/app.vue [hunk 3]
```diff
@@ -1,46 +1,79 @@
...
+                <v-btn icon="mdi-help-circle" @click="showHelpDialog = true" />
+                <v-btn icon="mdi-menu" @click="toggleDrawer" />
+
+                <HelpDialog v-model="showHelpDialog" />
+            </v-app-bar>
+
+            <!-- Navigation Drawer for the About section -->
+            <v-navigation-drawer v-model="drawer" location="right">
+                <v-list>
+                    <v-list-item @click="router.push('/about')" class="text-center">About</v-list-item>
```
Comment (alyxadarr) [inline wordle-web/app.vue:31]: Looks like you have an About section in the app bar but no about.vue page so it results in an error. either delete the v list item as we didn't absolutely need it for this assignment or make a quick about page!

File: wordle-web/app.vue [hunk 4]
```diff
@@ -1,46 +1,79 @@
...
+                    <v-list-item @click="setTheme('Merlin')" class="text-center">Merlin's Study</v-list-item>
+                </v-list>
+            </v-navigation-drawer>
+
+            <v-main>
+                <v-container class="text-center">
+                    <v-row>
+                        <v-col cols="12">
+                            <v-img src="https://media.giphy.com/media/l0ExsgrTuACbtPaqQ/giphy.gif"
+                                   alt="Animated spell casting"
```
Comment (alyxadarr) [inline wordle-web/app.vue:51]: Love the gif!

File: wordle-web/app.vue [hunk 5]
```diff
@@ -1,46 +1,79 @@
...
-        <template v-slot:prepend>
-          <v-icon color="secondary"> mdi-book </v-icon>
-        </template>
-
-        <v-app-bar-title>WordleApp</v-app-bar-title>
-
-        <v-btn @click="router.push('/')">Home</v-btn>
-        <v-btn @click="router.push('/test')">Test</v-btn>
-        <v-btn icon="mdi-theme-light-dark" @click="toggleTheme" />
-        <v-btn icon="mdi-help-circle" @click="showHelpDialog = true" />
```
Comment (alyxadarr) [inline wordle-web/app.vue:14]: We added another button here with a trophy mdi icon for the leaderboard page! Could give you a start

File: wordle-web/pages/index.vue [hunk 6]
```diff
@@ -41,17 +41,17 @@
...
+const game: Ref<Game> = ref(new Game("GAMES"));
 provide("GAME", game.value);
 
 const myGuess = ref("");
 
 onMounted(() => {
+  // Get random word from word list
+  getWordFromApi().then((word) => {
+    game.value = new Game(word);
+  });
```
Comment (alyxadarr) [inline wordle-web/pages/index.vue:53]: Be sure to add a leaderboard page leaderboard.vue once you get the implementation squared away with the controller! or get started on the bones of the page without implementation, we did this :)

File: Wordle.Api/Wordle.Api/Services/WordOfTheDayService.cs [hunk 7]
```diff
@@ -1,20 +1,51 @@
...
+    }
 
     public string GetRandomWord()
     {
-        if (Word == null)
+        Random random = new();
+        int index = random.Next(words.Count);
+        return words[index];
+    }
+
```
Comment (alyxadarr) [inline Wordle.Api/Wordle.Api/Services/WordOfTheDayService.cs:24]: You will also need a model cs file for the leaderboard implementation. This is where you add player and the relevant variables like PlayerId, Name, GameCount, and AverageAttempts per assignment

File: wordle-web/app.vue [hunk 8]
```diff
@@ -1,46 +1,79 @@
...
+                        Welcome to LexiQuest
+                    </v-btn>
+                </template>
+
+                <v-spacer></v-spacer> <!-- This pushes the rest of the items to the right -->
+
+                <v-btn icon="mdi-cog" @click="toggleSettingsDrawer" />
+                <v-btn icon="mdi-help-circle" @click="showHelpDialog = true" />
+                <v-btn icon="mdi-menu" @click="toggleDrawer" />
+
```
Comment (alyxadarr) [inline wordle-web/app.vue:21]: We created a dialogue for the player name and added that to our components folder. This resource helped me understand the "display the name as the user types" requirement! https://vuejs.org/guide/essentials/forms

File: .github/workflows/main_lexiquestapi.yml [hunk 9]
```diff
@@ -1,17 +1,19 @@
...
     branches:
       - main
+      - Assignment-3
     paths:
-      - Wordle.Api/**
+       - Wordle.Api/**
   pull_request:
     branches:
       - main
+      - Assingment-3
```
Comment (alyxadarr) [inline .github/workflows/main_lexiquestapi.yml:16]: spelling mistake ?

File: .github/workflows/main_lexiquestapi.yml [hunk 10]
```diff
@@ -63,7 +62,7 @@ jobs:
         id: deploy-to-webapp
```
Comment (alyxadarr) [inline .github/workflows/main_lexiquestapi.yml:62]: You might be missing some information in here, the deployment to azure was challenging for sure

**Review by alyxadarr — COMMENTED** [review]
It seems you were figuring out deployment and getting everything set up in azure! Good try overall, I will be excited to see what you implement! 

1. Create a leaderboard API Controller ❌

Create an endpoint that returns the top 10 scores (HttpGet) ❌
Results should include Name, average guesses, and number of games played ❌
Create an endpoint that allows for posting a score (HttpPost) with data in the body ❌
A new name should add a new record ❌
An existing name should update that record ❌
Save and load all data with Entity Framework to an Azure SQL database or LocalDb for debugging ❌
Class should be named "Player" and "Players" for the DbSet ❌
Fields should be:
int PlayerId ❌
string Name ❌
int GameCount ❌
double AverageAttempts ❌
int AverageSecondsPerGame ❌(OPTIONAL)
The logic should reside in a service that is injected via dependency injection into the controller ❌
Set up CORS (Cross Origin Resource Sharing) ✅
2. Create a dialog for getting the user's name ❌

If we don't have the user's name, present the user with a dialog that allows them to type in their name ❌
The users current name should show on the game page (upper right) ❌
This should be visible when setting/changing the name. It should change as the user types ❌
The user should be able to click on their name to change it (with the same dialog) ❌
Save the name in local storage and load it for the next game if it is set ❌
If the name is not set, display Guest as the name. ❌
Clicking on this allows for preemptive setting of the name before the end of the game ❌
Dialog must be persistent, doesn't close by clicking off the dialog ❌ just use "persistent" in the dialog class when you implement this !
3. Save the Score and Seconds to complete ❌

Save the score and time to complete in seconds at the end of the game if we have a name. ❌
If the user hasn't set a name, save the score and seconds as "Guest". (Be sure to prompt first) ❌
Save should be done with an Axios Post ❌
4. Create a Leaderboard Page ❌

Create a /leaderboard page in the application that shows the top scores ❌
Use Axios to call the leaderboard API ❌
Make a nice display of the returned data ❌
Add a link on the front page to access the leaderboard ❌
Add a link to the app bar to access the leaderboard ❌
5. Unit testing ❌ (OPTIONAL) (Not covered in class, but in instructor's repo: https://github.com/granterickson/EWU-CSCD379-2024-Spring)

Tests for the service ❌(OPTIONAL)
Integration tests for the controller ❌ (OPTIONAL)
6. Deploy site to Azure ✅

Deploy API site with Github Action to Azure Web App ✅
Unit Tests should run on build ❌ vitest was commented out so be sure to fix that 

Extra Credit
Unit testing on the client side (3) ❌
Animate the leaderboard (3) ❌
Create a logo (3) ❌ could def get some inspo from your theme to make a logo!
Add Average Seconds to win in score (3) ❌

File: wordle-web/app.vue (hunk 3 above)
Comment (Hex-Paladin) [reply wordle-web/app.vue:31]: Thanks, just basically took some scripts from assignment 2 for my general theme and layout, still need to go in and clean things up.

**Comment by SergeiAshikhmin** [comment]
I will review this :)

File: wordle-web/app.vue (hunk 3 above)
Comment (SergeiAshikhmin) [reply wordle-web/app.vue:31]: Confirm, the About page not working for me too.

**Review by SergeiAshikhmin — COMMENTED** [review]
Good code overall 👍 
There is a lot of job you have to do but great start.

I don't know about the logo, that is why I leave it unchecked.

## Instructions

**1. Create a leaderboard API Controller** ❌
- Create an endpoint that returns the top 10 scores (HttpGet) ❌
  - Results should include Name, average guesses, and number of games played ❌
- Create an endpoint that allows for posting a score (HttpPost) with data in the body ❌
  - A new name should add a new record ❌
  - An existing name should update that record ❌
- Save and load all data with Entity Framework to an Azure SQL database or LocalDb for debugging ❌
  - Class should be named "Player" and "Players" for the DbSet ❌
  - Fields should be: 
    - int PlayerId ❌
    - string Name ❌
    - int GameCount ❌
    - double AverageAttempts ❌
    - int AverageSecondsPerGame ❌(OPTIONAL)
- The logic should reside in a service that is injected via dependency injection into the controller ❌
- Set up CORS (Cross Origin Resource Sharing) ❌

**2. Create a dialog for getting the user's name** ❌
- If we don't have the user's name, present the user with a dialog that allows them to type in their name ❌
- The users current name should show on the game page (upper right) ❌
  - This should be visible when setting/changing the name. It should change as the user types ❌
- The user should be able to click on their name to change it (with the same dialog) ❌
- Save the name in local storage and load it for the next game if it is set ❌
- If the name is not set, display Guest as the name. ❌
  - Clicking on this allows for preemptive setting of the name before the end of the game ❌
- Dialog must be persistent, doesn't close by clicking off the dialog ❌

**3. Save the Score and Seconds to complete** ❌
- Save the score and time to complete in seconds at the end of the game if we have a name. ❌
- If the user hasn't set a name, save the score and seconds as "Guest". (Be sure to prompt first) ❌
- Save should be done with an Axios Post ❌

**4. Create a Leaderboard Page** ❌
- Create a /leaderboard page in the application that shows the top scores ❌
- Use Axios to call the leaderboard API ❌
- Make a nice display of the returned data ❌
- Add a link on the front page to access the leaderboard ❌
- Add a link to the app bar to access the leaderboard ❌

**5. Unit testing** ❌ (OPTIONAL) (Not covered in class, but in instructor's repo: https://github.com/granterickson/EWU-CSCD379-2024-Spring)
- Tests for the service ❌(OPTIONAL)
- Integration tests for the controller ❌ (OPTIONAL)

**6. Deploy site to Azure** ✅
- Deploy API site with Github Action to Azure Web App ✅
- Unit Tests should run on build ❌


## Extra Credit

- Unit testing on the client side (3) ❌
- Animate the leaderboard (3) ❌
- Create a logo (3) ❌✅
- Add Average Seconds to win in score (3) ❌

**Pull request closed without merging**